File: twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map<int, int> mapper;
        vector<int> res;
        int check;

        for(int i=0; i < nums.size(); i++) {
            check = target - nums[i];
            
            if(mapper.find(check) != mapper.end()){
                res.push_back(mapper[check]);
                res.push_back(i);         
                break;
            }
            mapper[nums[i]] = i;
        }
        return res;
        
    }
};

File: twoSum.py

# Leetcode - https://leetcode.com/problems/two-sum/


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        h = {}

        for i in range(0, len(nums)):

            val = target - nums[i]

            if val in h:
                return [h[val], i]
            else:
                h[nums[i]] = i
